e3nn/o3/_tensor_product/_codegen.py

```diff
@@ -39,14 +39,20 @@ def codegen_tensor_product_left_right(
     x2s = fx.Proxy(graph.placeholder("x2", torch.Tensor), tracer=tracer)
     weights = fx.Proxy(graph.placeholder("w", torch.Tensor), tracer=tracer)
 
-    empty = fx.Proxy(graph.call_function(torch.empty, ((),), dict(device="cpu")), tracer=tracer)
+    # Short circuit for zero-dimensional inputs
+    if len(irreps_in1) == 0 or len(irreps_in2) == 0:
+        if shared_weights:
+            output_shape = torch.broadcast_tensors(x1s[..., :1], x2s[..., :1])[0].shape[:-1]
```

Comment:
@mitkotak would this `:1` instead of `0` trick work if just applied below without even including the short circuit?